addons/makeslave.pl

```diff
@@ -2,36 +2,56 @@
 
 use Config::IniFiles;
 
+BEGIN {
+    # log4perl init
+    use constant INSTALL_DIR => '/usr/local/pf';
+    use lib INSTALL_DIR . "/lib";
+}
+
+use pf::file_paths qw($pf_config_file);
+
 my %ini;
 
 tie %ini, 'Config::IniFiles';
 
-my $pfconf = "/usr/local/pf/conf/pf.conf";
-
 my %inipfconf;
-tie %inipfconf, 'Config::IniFiles', ( -file => "$pfconf" );
+tie %inipfconf, 'Config::IniFiles', ( -file => "$pf_config_file" );
 
 my $replication_username = $inipfconf{"active_active"}{"galera_replication_username"};
 my $replication_password = $inipfconf{"active_active"}{"galera_replication_password"};
 
+if ($replication_username eq "" | $replication_password eq "") {
+    die "replication information are missing";
+}
 
 print "Enter the MySQL root password: ";
 my $mysql_root_password = <STDIN>;
 chomp $mysql_root_password;
 
-
-print "Enter the MySQL master ip address: ";
+print "Enter the MySQL master IP address: ";
 my $mysql_master_ip = <STDIN>;
 chomp $mysql_master_ip;
 
+if ($mysql_master_ip !~ /(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/) {
+    die "wrong ip address format";
+}
+
 $output = `sudo mysql -u root -p'$mysql_root_password' -h$mysql_master_ip -e "GRANT REPLICATION SLAVE ON *.*  TO '$replication_username'\@'%'"`;
 
+if ($output) {
```

Comment:
If there is an output it doesn't mean it failed.

I think you want to use system along with a check on the result or if you use the backticks to launch the command you should be checking $? and not the result of the command which is the output